src/components/sidebar/Mints/MintSettings.tsx

```diff
@@ -23,23 +35,28 @@ const MintSettings = () => {
 
    const walletConnected = useMemo(() => !!user.nwcUri, [user.nwcUri]);
 
+   useEffect(() => {
```

Comment:
we could avoid this useEffect buy invoking MintSettings component like this:
```
const { activeUnit } = useCashuContext()
...
<MintSettings key={activeUnit} />
```
in the siderbar. I wonder if we should do it though or keep as is

Reply:
probably fine to keep

Reply:
Fine for me to keep as is, but there's one other issue I wonder if this would help solve.

The problem is when you open and close the drawer, the state does not get reset.

At one point I had a conditional render of the entire drawer component, but looks like I removed that for some reason.

In SettingsSidebar I had:
```typescript
{!hidden && (
            <Drawer
                      ...
```

So that when the drawer closed it reset all the state. Is there a better way to do this? I would like all the dropdowns to close and for the active account tab to be reset.

Maybe pass onClose to the sidebar items?

Reply:
check it now pls. I've added key on drawer content